sensor/common/detector/filesystem/service/service_impl.go

```diff
@@ -0,0 +1,92 @@
+package service
+
+import (
+	"context"
+	"io"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/grpc/authz/idcheck"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/sensor/common/message"
+	"google.golang.org/grpc"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// NewService creates a new streaming service with the fact agent. It should only be called once.
+func NewService() Service {
+	srv := &serviceImpl{
+		authFuncOverride: authFuncOverride,
+		writer:           nil,
+	}
+
+	return srv
+}
+
+func authFuncOverride(ctx context.Context, fullMethodName string) (context.Context, error) {
+	err := idcheck.CollectorOnly().Authorized(ctx, fullMethodName)
+	return ctx, errors.Wrapf(err, "file activity authorization for %q", fullMethodName)
+}
+
+type serviceImpl struct {
+	sensor.UnimplementedFileActivityServiceServer
+
+	authFuncOverride func(context.Context, string) (context.Context, error)
+	writer           io.Writer
+}
+
+func (s *serviceImpl) Name() string {
+	return "filesystem.serviceImpl"
+}
+
+func (s *serviceImpl) Start() error {
+	log.Info("Start filesystem")
+	return nil
+}
+
+func (s *serviceImpl) Stop() {}
+
+func (s *serviceImpl) Capabilities() []centralsensor.SensorCapability {
+	return nil
+}
+
+func (s *serviceImpl) ResponsesC() <-chan *message.ExpiringMessage {
+	return nil
+}
```

Comment:
Are you planning on making this service a [SensorComponent](https://github.com/stackrox/stackrox/blob/ff75b2c3d60a257785549b453992123ae90a2ec0/sensor/common/component.go#L65-L77)? If not, then all these functions can be removed. If yes, then I'd argue we should not do that. We are aware that some services are also `SensorComponents` but we want to separate both concepts.